internal/provider/vxc_resource.go

```diff
@@ -0,0 +1,1494 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	megaport "github.com/megaport/megaportgo"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &vxcResource{}
+	_ resource.ResourceWithConfigure   = &vxcResource{}
+	_ resource.ResourceWithImportState = &vxcResource{}
+)
+
+// vxcResourceModel maps the resource schema data.
+type vxcResourceModel struct {
+	LastUpdated types.String `tfsdk:"last_updated"`
+
+	ID                 types.Int64  `tfsdk:"product_id"`
+	UID                types.String `tfsdk:"product_uid"`
+	ServiceID          types.Int64  `tfsdk:"service_id"`
+	Name               types.String `tfsdk:"product_name"`
+	Type               types.String `tfsdk:"product_type"`
+	RateLimit          types.Int64  `tfsdk:"rate_limit"`
+	DistanceBand       types.String `tfsdk:"distance_band"`
+	ProvisioningStatus types.String `tfsdk:"provisioning_status"`
+
+	SecondaryName  types.String `tfsdk:"secondary_name"`
+	UsageAlgorithm types.String `tfsdk:"usage_algorithm"`
+	CreatedBy      types.String `tfsdk:"created_by"`
+
+	ContractTermMonths types.Int64                   `tfsdk:"contract_term_months"`
+	CompanyUID         types.String                  `tfsdk:"company_uid"`
+	CompanyName        types.String                  `tfsdk:"company_name"`
+	Locked             types.Bool                    `tfsdk:"locked"`
+	AdminLocked        types.Bool                    `tfsdk:"admin_locked"`
+	AttributeTags      map[types.String]types.String `tfsdk:"attribute_tags"`
+	Cancelable         types.Bool                    `tfsdk:"cancelable"`
+
+	LiveDate          types.String `tfsdk:"live_date"`
+	CreateDate        types.String `tfsdk:"create_date"`
+	ContractStartDate types.String `tfsdk:"contract_start_date"`
+	ContractEndDate   types.String `tfsdk:"contract_end_date"`
+
+	PortUID           types.String              `tfsdk:"port_uid"`
+	AEndConfiguration *vxcEndConfigurationModel `tfsdk:"a_end"`
+	BEndConfiguration *vxcEndConfigurationModel `tfsdk:"b_end"`
+
+	AEndOrderConfiguration *vxcOrderEndPointConfigurationModel `tfsdk:"a_end_order_configuration"`
+	BEndOrderConfiguration *vxcOrderEndPointConfigurationModel `tfsdk:"b_end_order_configuration"`
```

Comment:
I would merge these into one type, so that there is only one place to edit fields like `InnerVLAN`. It's a little hard to merge that with the MVE/MCR destination options, but at the very least we shouldn't have an input and an output in the model.